src/value/loader.rs

```diff
@@ -43,9 +43,32 @@ pub trait Loader {
 
     /// Parses given Ion over a given slice into an [`Vec`] returning an
     /// [`IonError`](crate::result::IonError) if any error occurs during the parse.
+    #[inline]
     fn load_all(&self, data: &[u8]) -> IonResult<Vec<OwnedElement>> {
         self.iterate_over(data)?.collect()
     }
+
+    /// Parses given Ion over a given slice into a single [`Element`](super::Element) instance.
+    /// Returning [`IonError`](crate::result::IonError) if any error occurs during the parse
+    /// or there is more than one top-level [`Element`](super::Element) in the data.
+    #[inline]
+    fn load_one(&self, data: &[u8]) -> IonResult<OwnedElement> {
+        let mut iter = self.iterate_over(data)?;
+        match iter.next() {
+            Some(Ok(elem)) => {
+                // make sure there is nothing else
+                match iter.next() {
+                    None => Ok(elem),
+                    Some(Ok(_)) => {
+                        decoding_error("Expected a single element, but there was more than one")
+                    }
+                    Some(other) => other,
```

Comment:
This branch produces a potentially confusing `IonError`--it won't mention that a value was successfully parsed before invalid bytes were encountered. We should look into adding a source error field to the `IonError` enum variants so we can layer them when necessary.

Reply:
Added #222